File: app/android/app/src/main/java/com/jun/weather/repository/web/api/MidWeatherInfoService.java
package com.jun.weather.repository.web.api;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface MidWeatherInfoService {
    @GET("getMidLandFcst")
    Call<ApiResponse<MidLandItem>> getMidLandFcst(@QueryMap Map<String, Object> query);

    @GET("getMidTa")
    Call<ApiResponse<MidTempItem>> getMidTa(@QueryMap Map<String, Object> query);
}

File: app/android/app/src/main/java/com/jun/weather/repository/web/api/MidWeatherInfoService.kt
package com.jun.weather.repository.web.api

import com.jun.weather.repository.web.entity.ApiResponse
import com.jun.weather.repository.web.entity.MidLandItem
import com.jun.weather.repository.web.entity.MidTempItem
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.QueryMap

interface MidWeatherInfoService {
    @GET("getMidLandFcst")
    fun getMidLandFcst(@QueryMap query: Map<String, String>): Call<ApiResponse<MidLandItem>>

    @GET("getMidTa")
    fun getMidTa(@QueryMap query: Map<String, String>): Call<ApiResponse<MidTempItem>>
}